astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +188,44 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(PyUnitListProxy *self, PyUnitListProxy *other, int op)
+{int equal;
+  int status;
+   
+
+  if ((op == Py_EQ || op == Py_NE) &&
+      PyObject_TypeCheck(other, &PyUnitListProxyType)) {
+     
+
+    if (self->size!=other->size||self->unit_class!=other->unit_class||self->array!=other->array)
```

Comment:
I don't think this is correct to use pointer equality for `unit_class` or `array`. Are these singletons?

Reply:
I felt so that they are singletons. I will check it again